packages/router/src/grpc/view-protocol-server/note-by-commitment.ts

```diff
@@ -16,18 +13,13 @@ export const noteByCommitment: Impl['noteByCommitment'] = async (req, ctx) => {
 
   const noteByCommitment = await indexedDb.getSpendableNoteByCommitment(req.noteCommitment);
   if (noteByCommitment) return { spendableNote: noteByCommitment };
-  if (!req.awaitDetection) throw new ConnectError('Note not found', Code.NotFound);
 
   // Wait until our DB encounters a new note with this commitment
-  const response = new NoteByCommitmentResponse();
-  const subscription = indexedDb.subscribe('SPENDABLE_NOTES');
-
-  for await (const update of subscription) {
-    const note = SpendableNoteRecord.fromJson(update.value);
-    if (note.noteCommitment?.equals(req.noteCommitment)) {
-      response.spendableNote = note;
-      break;
+  if (req.awaitDetection) {
+    for await (const update of indexedDb.subscribe('SPENDABLE_NOTES')) {
+      const spendableNote = SpendableNoteRecord.fromJson(update.value);
+      if (spendableNote.noteCommitment?.equals(req.noteCommitment)) return { spendableNote };
     }
   }
-  return response;
+  throw new ConnectError('Note not found', Code.NotFound);
```

Comment:
I noticed that `swapByCommitment` and `noteByCommitment` differently handled the case when `awaitDetection` is true but no record was detected. 
I think we should throw error rather than an return empty response in such a case